arm-sql/2015-05-01-preview/swagger/virtualNetworkRules.json

```diff
@@ -0,0 +1,277 @@
+          },
+          "default": {
+            "description": "*** Error Responses: ***\n\n * 400 InvalidResourceId - Invalid resource identifier.\n\n * 400 MismatchingSubscriptionWithUrl - The provided subscription did not match the subscription in the Url.\n\n * 400 MismatchingResourceGroupNameWithUrl - The provided resource group name did not match the name in the Url.\n\n * 400 MismatchingServerNameWithUrl - The provided server name did not match the name in the Url.\n\n * 400 NullVirtualNetworkRequest - Virtual Network Request is Null\n\n * 400 NullVirtualNetworkRequestParameters - Virtual Network Request Parameters are Null\n\n * 404 SubscriptionDoesNotHaveServer - The requested server was not found\n\n * 404 OperationIdNotFound - The operation with Id does not exist.\n\n * 409 OperationCancelled - The operation has been cancelled by user.\n\n * 409 OperationInterrupted - The operation on the resource could not be completed because it was interrupted by another operation on the same resource.\n\n * 500 OperationTimedOut - The operation timed out and automatically rolled back. Please retry the operation."
+          },
+          "204": {
+            "description": "The specified Vnet Firewall Rule does not exist."
+          }
+        },
+        "x-ms-long-running-operation": true
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/servers/{serverName}/virtualNetworkRules": {
+      "get": {
+        "tags": [
+          "VnetFirewallRule"
+        ],
+        "description": "Returns information about Vnet Firewall Rules.",
+        "operationId": "VnetFirewallRules_List",
+        "x-ms-examples": {
+          "Lists the virtual network firewall rules": { "$ref": "../examples/VirtualNetworkRulesList.json" }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupParameter"
+          },
+          {
+            "$ref": "#/parameters/ServerNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully retrieved the list of vnet firewall rules.",
+            "schema": {
+              "$ref": "#/definitions/VnetFirewallRuleList"
+            }
+          },
+          "default": {
+            "description": "*** Error Responses: ***\n\n * 404 SubscriptionDoesNotHaveServer - The requested server was not found\n\n * 404 ResourceNotFound - The requested resource was not found."
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "VnetFirewallRule": {
+      "description": "Vnet Firewall Rule Resource.",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/VnetFirewallRuleResourceProperties"
+        },
+        "location": {
```

Comment:
Looks like a Resource, there are a couple of options here either reference [this](https://github.com/Azure/azure-rest-api-specs/blob/master/arm-sql/2015-05-01-preview/swagger/blobAuditingPolicies.json#L123) model or redefine the same structure here and do an allOf. (I'd recommend not redefining simply since its duplication and overhead)